prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -1346,6 +1347,84 @@ static int efa_rdm_ep_set_shared_memory_permitted(struct efa_rdm_ep *ep, bool sh
 	return 0;
 }
 
+/**
+ * @brief Conditionally set efa_rdm_ep#max_msg_size per user's request
+ *
+ * If the requested msg size exceeds the EFA provider's default value, the
+ * request is rejected.
+ *
+ * @param[in,out]	ep		EFA RDM endpoint
+ * @param[in]		max_msg_size	Requested maximum msg size
+ *
+ * @return		0 on success, -FI_EINVAL otherwise
+ *
+ * @sa #FI_OPT_MAX_MSG_SIZE
+ */
+static int efa_rdm_ep_set_max_msg_size(struct efa_rdm_ep *ep, size_t max_msg_size)
+{
+	if (max_msg_size > ep->max_msg_size) {
```

Comment:
This needs to be modified s.t. `max_msg_size` is compared against the EFA provider's static limit. Otherwise, the application would be unable to increase the max size via `fi_setopt()` after reducing it.